app/models/job/state_machine.rb

```diff
@@ -51,4 +65,35 @@ def queue_signal(*args, priority: MiqQueue::NORMAL_PRIORITY, role: nil, deliver_
       :server_guid => server_guid
     )
   end
+
+  #
+  # Helper methods for display of transitions
+  #
+
+  def to_dot(include_starred_states = false)
+    all_states = transitions.values.map(&:to_a).flatten.uniq.reject { |t| t == "*" }
+
+    "".tap do |s|
+      s << "digraph #{self.class.name.inspect} {\n"
+      transitions.each do |signal, signal_transitions|
+        signal_transitions.each do |from, to|
+          next if !include_starred_states && (from == "*" || to == "*")
+
+          from = from == "*" ? all_states : [from]
+          to   = to == "*"   ? all_states : [to]
+          from.product(to).each { |f, t| s << "  #{f} -> #{t} [ label=\"#{signal}\" ]\n" }
+        end
+      end
+      s << "}\n"
+    end
+  end
+
+  def to_svg(include_starred_states = false)
+    require "open3"
+    out, err, _status = Open3.capture3("dot -Tsvg", :stdin_data => to_dot(include_starred_states))
```

Comment:
Does this require any Gemfile changes or anything to be preinstalled?  Assuming these are for developers only, and will not cause issues in production.

Reply:
graphviz needs to be installed for this method to work, otherwise you get an Exception (which is expected).  This is for devs and should not cause issues in production.  I probably *could* raise an exception when `!Rails.env.development?`, but that's just trading one exception for another.

Reply:
yeah - no need to trade one exception for another - especially since nothing is calling this method in the code